cub/cub/agent/agent_adjacent_difference.cuh

```diff
@@ -36,6 +36,9 @@
 
 #include <thrust/system/cuda/detail/core/util.h>
 
+#ifdef CCCL_AS_SYSTEM_HEADER
```

Comment:
question: Would it be better to put this _before_ all the `#includes`? That way if we include a header that inadvertently doesn't already use `CCCL_AS_SYSTEM_HEADER`, then it will still silence warnings from that header.

suggestion: We could standardize always using `CCCL_AS_SYSTEM_HEADER` immediately after the `#pragma once`. This would make it easier to build an automation that verifies every header has this annotation.

Reply:
No, we do need the define from somewhere, so we would always need to include at least one config header first.

Furthermore, this would just hide bugs in our implementation, so I am not a fan of that

Reply:
> No, we do need the define from somewhere, so we would always need to include at least one config header first.
Ah, right, so maybe then
```
#pragma once
#include <cuda/__config>
CCCL_AS_SYSTEM_HEADER
// rest of includes
```

It would not hide bugs in our implementation as we would run our tests with this macro disabled. Furthermore, we've had issues with warnings generated in user code from headers we don't control, like MSVC's STL headers.

We want to **guarantee** that users will not see warnings from our headers by default.